### PR TITLE
Add SimHitMergeAlg and update G2CDArborAlg

### PR DESCRIPTION
**Pull request by wenxingfang**
Hi @mirguest @LinghuiWuIHEP

I add one alg (SimHitMergeAlg) for merging simcalorimeterhit from simulation by cell id. Now the G2CDArborAlg can use the data after SimHitMergeAlg for processing digitization for CEPCSW simulated data, just like processing simulation data produced by Mokka.
Besides, I correct the G2CDArborAlg a bit (mainly add a selection of readout, modify the layer number). Could you review the changes?

Best,
Wenxing

### CODE REVIEW

**Comment by LinghuiWuIHEP** [comment]
Hi @mirguest , I have checked the commit. Could you merge it?

File: Digitisers/SimHitMerge/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+gaudi_subdir(SimHitMergeAlg v0r0)
+
+find_package(DD4hep COMPONENTS DDG4 REQUIRED)
+find_package(EDM4HEP REQUIRED)
+find_package(podio REQUIRED ) 
+find_package(K4FWCore REQUIRED)
```
Comment (mirguest) [inline Digitisers/SimHitMerge/CMakeLists.txt:6]: Please rename K4 to k4. Thanks.

File: Examples/options/LCIO_read_G2CD.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+
+from Gaudi.Configuration import *
+from Configurables import K4DataSvc
+dsvc = K4DataSvc("EventDataSvc")
```
Comment (mirguest) [inline Examples/options/LCIO_read_G2CD.py:5]: Please also rename it to k4.

Comment (wenxingfang) [reply Examples/options/LCIO_read_G2CD.py]: Hi @mirguest ，

Thanks for the comments, I have updated the pull request.

Best,
Wenxing

**Review by mirguest — APPROVED** [review]
@wenxingfang , looks good. Thanks.
Tao

**Pull request merged**